### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by AmaliaGM**
## Project requirements

- Create a file named schema_based_on_diagram.sql where you implement the database from the diagram.
- Join tables from many-to-many relationships might not appear in the diagram, but you still need them.
- Remember to add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        REFERENCES public.treatments (id) MATCH SIMPLE
+        ON UPDATE NO ACTION
+        ON DELETE NO ACTION
+        NOT VALID
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.invoice_items
+    OWNER to postgres;
```
Comment (ShahierNashaat) [inline schema_based_on_diagram.sql]: - Kindly create a table for the relation many-to-many between medical_histories and treatments.

```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
);
```

- [ ] Also create indexes for all FOREIGN KEYS have indexes created.
* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @AmaliaGM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (AmaliaGM) [reply schema_based_on_diagram.sql]: @ShahierNashaat Hi. Thanks for your review. There's a relationship between them. You mean another?

CREATE TABLE IF NOT EXISTS public.medical_histories
(
    id integer NOT NULL,
    admitted_at timestamp without time zone,
    patient_id integer,
    status character varying COLLATE pg_catalog."default",
    CONSTRAINT medical_histories_pkey PRIMARY KEY (id),
    CONSTRAINT id FOREIGN KEY (id)
        **REFERENCES public.treatments (id) MATCH SIMPLE**
        ON UPDATE NO ACTION
        ON DELETE NO ACTION
        NOT VALID,
    status character varying COLLATE pg_catalog."default",
    CONSTRAINT medical_histories_pkey PRIMARY KEY (id),
    CONSTRAINT id FOREIGN KEY (id)

Comment (ShahierNashaat) [reply schema_based_on_diagram.sql]: Yes another relation

Comment (AmaliaGM) [reply schema_based_on_diagram.sql]: Hi. Sorry to disagree with you. This are the two diagrams (from the github project and from my table). Please compare.
Medical_Database_Diagram pgerd
clinic_diagram

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    CONSTRAINT patients_pkey PRIMARY KEY (id)
+)
+
+TABLESPACE pg_default;
+
+ALTER TABLE IF EXISTS public.patients
+    OWNER to postgres;
+	
+CREATE TABLE IF NOT EXISTS public.medical_histories
+(
```
Comment (BenMukebo) [inline schema_based_on_diagram.sql:31]: - [ ] As requested by the previous reviewer, Kindly make sure all  "join table" for the many-to-many relationships is created, even if they are not shown on the diagram. Therefore you should create a table between `medical_histories` and `treatments` (name can be different):

  ```sql
    CREATE TABLE medical_histories_has_treatments (
      medical_history_id int refrences medical_histories(id),
      treatment_id int refrences treatments(id),
      );
  
  ```
* Kindly address this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @AmaliaGM, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @AmaliaGM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**